lib/chef/resource/windows_certificate.rb

```diff
@@ -58,8 +58,9 @@ class WindowsCertificate < Chef::Resource
 
       action :create do
         description "Creates or updates a certificate."
-
-        add_cert(OpenSSL::X509::Certificate.new(raw_source))
+        converge_by("Adding certificate #{new_resource.source} from Store #{new_resource.store_name}") do
```

Comment:
Without any sort of additional logic this will cause the resource to update on every chef run. You need an if statement like you have below on delete

Reply:
@tas50 I have removed the converge_by block for create action and address in different ticket to address idempotency.